ModelBundle/Tests/EventListener/GenerateIdListenerTest.php

```diff
@@ -130,7 +130,7 @@ public function provideAnnotations()
 
         return array(
             array($repository0, $annotations0, $document0, 'fakename'),
-            array($repository1, $annotations1, $document1, 'aaaaaceeeeiiiinooooouuuuyyaaaaaceeeeiiiinooooouuuuy%20%5C'),
+            array($repository1, $annotations1, $document1, 'aaaaaceeeeiiiinooooouuuuyyaaaaaceeeeiiiinooooouuuuy__'),
```

Comment:
I think the result should be trimed in order to dismiss the replacement character to be starting or ending the result. I think consecutive replacement chars should also be avoided and replaced with only one instance.
We should also have some test cases for ponctuation marks.
Some tests to add for this behavior :
"toto ?" => "toto"  (not "toto__" as the result is now)
"tata..toto" => "tata_toto" (not tata__toto)